Docs update for KEP-5517 alpha2 (#56318)
kubernetes/website: content/en/docs/concepts/resource-management/dynamic-resource-allocation/how-dra-works.md

in the `kube-apiserver` and `kube-scheduler`.

> ## Node allocatable resources {#node allocatable-resources}

Reviewer: looks like a search-and-replace error? I think the anchor needs the hyphen?

Author: Thanks. Fixed.

When authoring a PodSpec using claims for these types of devices, there are a few things to be aware of:

*   When Pod-level resources are used, the scheduler strictly validates them against both container requests and limits:
    *  The sum of all container requests and DRA claim resources must not exceed the Pod-level requests; otherwise, the Pod will fail to schedule.
    *  Each individual container's limit plus its DRA allocations must not exceed the Pod-level limits; otherwise, the Pod will fail to schedule.
*   A container's total resource requirement is the sum of its container-level resources
    and any node allocatable resources from its associated resource claims.

> *   Claims that consume node allocatable resources with `Mapping` cannot be shared between Pods.

Reviewer 2: "Mapping" here is a bit ambiguous until the context explaining the two use cases below. Could we make that association a bit more explicit, e.g. something like

```suggestion
*   Claims that consume node allocatable resources defined by a `mapping` cannot be shared between Pods.
```

Then `mapping` at least matches the API field name.

    attributes:
      example.com/model:
        string: "model-x"
    nodeAllocatableResources:
      overhead:
        memory:
          perPod: "8Gi"
```

After a Pod is successfully bound to the node, the exact quantities of 

> node allocatable resources allocated via DRA are aggregated by the Kube-Scheduler

Reviewer 2: style nit: https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names

```suggestion
node allocatable resources allocated via DRA are aggregated by the `kube-Scheduler`
```

and embedded directly into the Pod's `status.nodeAllocatableResourceClaimStatuses` field.
This provides a clear, persistent handoff from the scheduler to the kubelet.

> Crucially, the Kubelet natively consumes this API to perfectly align system-level boundaries:

Reviewer 2: Similarly,

```suggestion
Crucially, the `kubelet` natively consumes this API to perfectly align system-level boundaries:
```

> - **Cgroups**: Pod and container cgroups would now include DRA based allocations, preventing workloads from being artificially throttled by the kernel.

Reviewer 2: nit: [Linux docs](https://docs.kernel.org/admin-guide/cgroup-v2.html#terminology) don't capitalize "cgroups".

> - **OOM Scores**: The Kubelet factors the container's DRA memory requests into its effective memory request.

Reviewer 2: ```suggestion
- **OOM Scores**: The `kubelet` factors the container's DRA memory requests into its effective memory request.
```

Node allocatable resources is an alpha feature and is enabled when the
[`DRANodeAllocatableResources` feature gate](/docs/reference/command-line-tools-reference/feature-gates/#DRANodeAllocatableResources) is enabled in the kube-apiserver,

> kube-scheduler, and kubelet.

Reviewer 2: ```suggestion
[`DRANodeAllocatableResources` feature gate](/docs/reference/command-line-tools-reference/feature-gates/#DRANodeAllocatableResources) is enabled in the `kube-apiserver`,
`kube-scheduler`, and `kubelet`.
```
